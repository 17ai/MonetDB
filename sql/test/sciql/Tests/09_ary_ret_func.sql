CREATE FUNCTION test_fixed ()
RETURNS ARRAY (x INT DIMENSION, y INT DIMENSION, val FLOAT)
BEGIN
	DECLARE ARRAY a (x INT DIMENSION[5], y INT DIMENSION[3], val FLOAT DEFAULT 42.0);
	RETURN a;
END;
SELECT * FROM test_fixed();
DROP FUNCTION test_fixed;

-- CREATE FUNCTION test_param (sizex INT, sizey INT, dfl_val FLOAT)
-- RETURNS ARRAY (x INT DIMENSION, y INT DIMENSION, val FLOAT)
-- BEGIN
-- 	DECLARE ARRAY a (x INT DIMENSION[sizex], y INT DIMENSION[sizey], val FLOAT DEFAULT dfl_val);
-- 	RETURN a;
-- END;
-- SELECT * FROM test_param (5, 3, 22.4);
-- DROP FUNCTION test_param;

