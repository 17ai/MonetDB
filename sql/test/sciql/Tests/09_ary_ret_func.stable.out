stdout of test '09_ary_ret_func` in directory 'test/sciql` itself:


# 18:49:56 >  
# 18:49:56 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/export/scratch2/zhang/monet-install/sciql/debug/var/MonetDB" "--set" "mapi_open=true" "--set" "mapi_port=35684" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_test_sciql" "--set" "mal_listing=0"
# 18:49:56 >  

# MonetDB 5 server v11.10.0
# This is an unreleased version
# Serving database 'mTests_test_sciql', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.629 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2012 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://riga.ins.cwi.nl:35684/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_history.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_compress.sql
# loading sql script: 18_dictionary.sql
# loading sql script: 19_cluster.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 24_zorder.sql
# loading sql script: 25_debug.sql
# loading sql script: 29_array.sql
# loading sql script: 39_analytics.sql
# loading sql script: 40_geom.sql
# loading sql script: 80_udf.sql
# loading sql script: 99_system.sql

# 18:49:56 >  
# 18:49:56 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=riga" "--port=35684"
# 18:49:56 >  

#CREATE FUNCTION test_fixed ()
#RETURNS ARRAY (x INT DIMENSION, y INT DIMENSION, val FLOAT)
#BEGIN
#	DECLARE ARRAY a (x INT DIMENSION[5], y INT DIMENSION[3], val FLOAT DEFAULT 42.0);
#	RETURN a;
#END;
#SELECT * FROM test_fixed();
% .,	.,	. # table_name
% x,	y,	val # name
% int,	int,	double # type
% 1,	1,	24 # length
[ 0,	0,	42	]
[ 0,	1,	42	]
[ 0,	2,	42	]
[ 1,	0,	42	]
[ 1,	1,	42	]
[ 1,	2,	42	]
[ 2,	0,	42	]
[ 2,	1,	42	]
[ 2,	2,	42	]
[ 3,	0,	42	]
[ 3,	1,	42	]
[ 3,	2,	42	]
[ 4,	0,	42	]
[ 4,	1,	42	]
[ 4,	2,	42	]
#DROP FUNCTION test_fixed;

# 18:49:56 >  
# 18:49:56 >  "Done."
# 18:49:56 >  

